### PR TITLE
subagents start

### PR DESCRIPTION
**Pull request by ncfrey**
## Description
This pull request introduces two new specialized agents, `code-reviewer` and `test-suite-architect`, aimed at improving code review quality and test coverage. Each agent is tailored for specific tasks, with detailed guidelines and best practices to ensure high standards in their respective areas.

#### Code Review Enhancements:
* Added the `code-reviewer` agent, designed to provide comprehensive code reviews with a focus on correctness, maintainability, performance, security, architecture, testing, documentation, and adherence to project standards. It includes a structured feedback approach and emphasizes constructive, educational communication. (`.claude/agents/code-reviewer.md`, [.claude/agents/code-reviewer.mdR1-R51](diffhunk://#diff-c0a2150c92b282c9817c6c5f7ebaec0c020d45e9aeedff28f8c49e21c09cafebR1-R51))

#### Test Coverage Improvements:
* Introduced the `test-suite-architect` agent, specialized in designing and implementing robust test suites. It covers unit tests, integration tests, end-to-end tests, and performance tests, with a focus on edge case identification, test isolation, and framework adaptability. (`.claude/agents/test-suite-architect.md`, [.claude/agents/test-suite-architect.mdR1-R66](diffhunk://#diff-c108701158dbbb717d5ddd7dc7e97ae0563c9d7eaa0464c31bddde11542b7dfaR1-R66))

## Type of Change
- [ ] Bug fix
- [X] New feature  
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

### CODE REVIEW

**Review by taylormjs — COMMENTED** [review]
lgtm! I bet a Principal/Staff software engineer would give us better than a B+ b/c they know about fasta files

File: .claude/agents/test-suite-architect.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+- Use descriptive test names that clearly indicate what is being tested
+- Follow the Arrange-Act-Assert pattern for clarity
+- Create reusable test fixtures and utilities
+- Implement proper mocking and stubbing for external dependencies
+- Include both positive and negative test cases
+
+**Edge Case Identification**: Systematically identify and test:
+- Boundary conditions and limit cases
+- Invalid inputs and error scenarios
+- Race conditions and concurrency issues
```
Comment (taylormjs) [inline .claude/agents/test-suite-architect.md:30]: Great idea to check for race conditions. This came up during an earlier training run with the lit datamodule btw

File: .claude/agents/code-reviewer.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+2. Understand the context and purpose of the code within the broader system
+3. Consider both immediate functionality and long-term maintainability
+4. Evaluate adherence to project-specific patterns from CLAUDE.md context
+
+**REVIEW FOCUS AREAS:**
+- **Correctness**: Identify bugs, logic errors, edge cases, and potential runtime issues
+- **Code Quality**: Assess readability, maintainability, and adherence to Python best practices
+- **Performance**: Spot inefficiencies, memory leaks, and optimization opportunities
+- **Security**: Check for potential vulnerabilities and unsafe operations
+- **Architecture**: Evaluate design patterns, separation of concerns, and code organization
```
Comment (taylormjs) [inline .claude/agents/code-reviewer.md:23]: Anecdotally, architecture is the area where Cursor and other coding agents have struggled the most and caused the most headaches. I'm curious to see how much this will refactor MRs
